### PR TITLE
rename sdk function

### PR DESCRIPTION
**Pull request by kristifidani**
# Pull Request Template

## Motivation and Context

Explain why this change is necessary or the issue it addresses.

## Summary

Renamed sdk `set_password` so it's easier to understand for users. This was mostly find all and replace.

- Rename `set_password` to `set_wallet_password`.
- Rename bindings `setPassword` to `setWalletPassword`

- Rename also `is_password_set` to `is_wallet_password_set`.
- Rename bindings `isPasswordSet` to `isWalletPasswordSet`.

- Updated the links in the API reference table.

## Checklist

Please ensure that your PR meets the following requirements:

### General

- [ ] Code follows project style and guidelines.
- [ ] No redundant or duplicate code.
- [ ] No added new dependencies without clear justification and approval.

### Documentation

- [ ] Added/updated relevant documentation (e.g., README, inline comments).
- [ ] Added a CHANGELOG entry for major changes.

### Testing

- [ ] Added relevant test cases, leveraging tools like `rstest` or similar for reusable tests.
- [ ] Tests pass locally.

### Build & CI/CD

- [ ] Confirmed no CI/CD pipeline issues.
- [ ] Updated build scripts where necessary.

## Comments to Reviewer

Please provide any comments or points of consideration for the reviewers, such as specific areas of focus, potential edge cases, or context that would aid the review process.

### Reviewer Responsibilities

- [ ] Code adheres to style and guidelines.
- [ ] All tests are appropriately covered and they pass.
- [ ] Consider performance and/or security impacts.
- [ ] Constructive feedback is provided.

### CODE REVIEW

File: sdk/docs/SDK Reference/SDK API Reference.md [hunk 1]
```diff
@@ -450,26 +450,26 @@ The Rustdoc API reference is available [here](../rust-docs/doc/sdk/index.html).
 | Set the password for the wallet using the provided pin and new password. If the password is already set, this changes it to the new password. Use [`is_password_set`](./SDK%20API%20Reference.md#is-password-set) to check if the password is already set. | `pin` - The new or existing PIN for the wallet, `new_password` - The new password to set for the wallet | [Wallet initialization](./SDK%20API%20Reference.md#create-new-wallet) | Usage | Application |
```
Comment (lszetg) [inline sdk/docs/SDK Reference/SDK API Reference.md:450]: > [`is_password_set`](./SDK%20API%20Reference.md#is-password-set)
maybe we could also change `is_password_set` to `is_wallet_password_set`? What do you think? to keep it consistent

File: sdk/bindings/android/src/lib.rs [hunk 2]
```diff
@@ -775,19 +775,19 @@ mod ffi {
...
         });
         result.map_err(|e| format!("{e:#?}"))
     }
 
     /// Check if the password to use for wallet operations is set.
-    /// Use {@link #setPassword} to set a new or change an existing password.
+    /// Use {@link #setWalletPassword} to set a new or change an existing password.
     ///
     /// @return whether the password is already set or not.
     #[public_name = "isPasswordSet"]
```
Comment (lszetg) [inline sdk/bindings/android/src/lib.rs:793]: somewhere below I posted question about it (`isPasswordSet` -> `isWalletPasswordSet`)

**Review by lszetg — COMMENTED** [review]
nice, just pointed out few places where we could consider applying changes to keep wallet context `Password` => `WalletPassword`

File: sdk/docs/SDK Reference/SDK API Reference.md [hunk 3]
```diff
@@ -450,26 +450,26 @@ The Rustdoc API reference is available [here](../rust-docs/doc/sdk/index.html).
...
-    public func setPassword(pin: String, new_password: String) throws
+    public func setWalletPassword(pin: String, new_password: String) throws
     ```
 
 ### Is password set
 
 | Method | Arguments | Returns | Dependencies | Level | Repeat|
 |--------|-----------|---------|--------------|-------|-------|
-| Check if the password is already set. Useful to prompt the user to setup one if it has not yet been done. See also [`set_password`](./SDK%20API%20Reference.md#set-password) for how to set a new password and change an existing password. | | Returns `true` or `false` whether or not the password is already set. | [Wallet initialization](./SDK%20API%20Reference.md#create-new-wallet) | Usage | Application |
+| Check if the password is already set. Useful to prompt the user to setup one if it has not yet been done. See also [`set_wallet_password`](./SDK%20API%20Reference.md#set-password) for how to set a new password and change an existing password. | | Returns `true` or `false` whether or not the password is already set. | [Wallet initialization](./SDK%20API%20Reference.md#create-new-wallet) | Usage | Application |
```
[the remark this answers is not in the document]
Comment (kristifidani) [reply sdk/docs/SDK Reference/SDK API Reference.md:472]: Yes, good catch 👍 . Updated it to `./SDK%20API%20Reference.md#set-wallet-password`

**Review by lszetg — APPROVED** [review]
🎉

**Review by antbern — APPROVED** [review]
